### PR TITLE
feat: [PIPE-24059]: [PIPE-PIPE-24186]: pipeline and execution list pages

### PR DESCRIPTION
**Pull request by srdjan-harness**
Screenshot 2024-12-18 at 21 17 22
Screenshot 2024-12-18 at 21 17 05

### CODE REVIEW

File: packages/ui/src/components/no-search-results.tsx
Comment (shaurya-harness) [inline packages/ui/src/components/no-search-results.tsx:1]: is this component needed?

File: packages/ui/src/views/pipelines/pipeline-list/pipeline-list-page.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  }, [])
+
+  if (isError)
+    return (
+      <>
+        <SandboxLayout.Main hasHeader hasLeftPanel>
+          <SandboxLayout.Content>
+            <Spacer size={2} />
+            <Text size={1} className="text-destructive">
+              {errorMessage || 'Something went wrong'}
```
Comment (shaurya-harness) [inline packages/ui/src/views/pipelines/pipeline-list/pipeline-list-page.tsx:43]: there should be a nodata state here I guess? with description as the errormessage

File: packages/ui/src/components/no-search-results.tsx
Comment (srdjan-harness) [reply packages/ui/src/components/no-search-results.tsx:1]: deleted

File: packages/ui/src/views/pipelines/execution-list/execution-list.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+  if (isLoading) {
+    return <SkeletonList />
+  }
+
+  if (noData) {
+    return query ? (
+      <StackedList.Root>
+        <div className="flex min-h-[50vh] items-center justify-center py-20">
+          <NoData
```
[the remark this answers is not in the document]
Comment (srdjan-harness) [reply packages/ui/src/views/pipelines/execution-list/execution-list.tsx:71]: its used in old pages, not in new

File: apps/gitness/src/components/LinkComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { Link } from 'react-router-dom'
+
+import { TLinkComponent } from '@harnessio/ui/views'
+
+export const LinkComponent: TLinkComponent = ({ to, children }: { to: string; children: React.ReactNode }) => (
```
Comment (shaurya-harness) [inline apps/gitness/src/components/LinkComponent.tsx:5]: not blocking, but let's add a disabled prop to this component to have pointer events as none.

File: apps/gitness/src/pages-v2/repo/repo-execution-list.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        pipelineId={pipelineId}
+        branch={executions && executions.length > 0 ? executions[0].source : undefined} // TODO: check this
+        toExecutions={'./executions'}
+      /> */}
+    </>
+  )
+}
+
+// NOTE: consider move this function to another file/location
+function apiExecutions2Executions(data: ListExecutionsOkResponse): IExecution[] {
```
Comment (shaurya-harness) [inline apps/gitness/src/pages-v2/repo/repo-execution-list.tsx:89]: not blocking, I have created a transform-utils folder, we can keep all of this stuff there for related pages

**Review by shaurya-harness — APPROVED** [review]

**Pull request merged**